### PR TITLE
Format 'source' And 'test' Files

### LINKED ISSUE #158: Format skips test files
I love that this project formatting built-in! I'm using Mint `0.7.1` on macOS (installed today via brew), and the only files in my project are what I get with `mint-lang init <name>`.

As a test I completely out-dended both `source/Main.mint` and `tests/Main.mint`, then ran `mint-lang format`:

```
➤ mint-lang format **/*.mint
Mint - Formatting files
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Formatted: source/Main.mint
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
All done in 1.367ms!
```

I also tried `mint-lang format source/** tests/**` and `mint-lang format {source,tests}/**`, but each only formatted `./source`.

I was only able to format `./tests` with `mint-lang format tests/**`, but I'd really prefer to format all Mint files in one go.

### PR DESCRIPTION
**Pull request by s0kil**
Fixes #158

### CODE REVIEW

**Comment by s0kil** [comment]
Should we respect the `source-directories` and `test-directories` configuration?

**Comment by gdotdesign** [comment]
I think it's a good idea, certainly better than using hard coded paths.

**Comment by s0kil** [comment]
It looks good to go.

File: src/commands/format.cr [hunk 1]
```diff
@@ -6,12 +6,26 @@ module Mint
...
+          format_directories_patterns =
+            format_directories.map do |dir|
+              File.join(dir, "**/*.mint")
+            end
+
+          if arguments.pattern.to_s.empty?
+            files = Dir.glob(format_directories_patterns)
+          else
+            files = Dir.glob(arguments.pattern.to_s)
+          end
```
Comment (Sija) [inline src/commands/format.cr]: One pedantic detail - what about this:

```cr
file_pattern = arguments.pattern.to_s.presence || format_directories_patterns
files = Dir.glob(file_pattern)
```

Comment (s0kil) [reply src/commands/format.cr]: I like that. What is `presence` in this context?

Comment (Sija) [reply src/commands/format.cr]: See [`String#presence`](https://crystal-lang.org/api/0.32.1/String.html#presence:self?-instance-method) and [`Nil#presence`](https://crystal-lang.org/api/0.32.1/Nil.html#presence-instance-method)

Comment (s0kil) [reply src/commands/format.cr]: Thank You, What is the relationship with `?`?

Comment (Sija) [reply src/commands/format.cr]: I'm sorry, I don't get the question...

Comment (Sija) [reply src/commands/format.cr]: I'm sorry, I don't get the question...

Comment (s0kil) [reply src/commands/format.cr]: Does it make sense to do just `presence` or `present?` ?

Comment (Sija) [reply src/commands/format.cr]: `presence` is the name.

Comment (s0kil) [reply src/commands/format.cr]: Got it.

Comment (s0kil) [reply src/commands/format.cr]: Tried it, turned into a crazy error message, ha
https://pastebin.com/fkjgJ2r4

Comment (s0kil) [reply src/commands/format.cr]: I don't want to spend much time on something so insignificant.

**Comment by gdotdesign** [comment]
Thanks :+1:

**Pull request merged**